component.js

```diff
@@ -95,19 +95,21 @@ module.exports.withDefaults = factory;
 function factory (options) {
   var debug;
   options = options || {};
-  var _shouldComponentUpdate = options.shouldComponentUpdate;
+  var _componentWillMount = options.componentWillMount ||
+                            componentWillMount;
+  var _shouldComponentUpdate = options.shouldComponentUpdate ||
+                               shouldComponentUpdate.withDefaults(options);
+  var _componentWillReceiveProps = options.componentWillReceiveProps ||
```

Comment:
I don't know if we want to have this as an overriable. Would it be necessary to deactivate this behaviour? if not, instead of overriding the mixin, you would simply pass in a new mixin as all life cycle mixins are guaranteed to be triggered. Or does ordering matter in this case?